nixos/tests/networking.nix

```diff
@@ -368,6 +368,78 @@ let
           $router->waitUntilSucceeds("ping -c 1 192.168.1.3");
         '';
     };
+    ipvlan = {
+      name = "IPVLAN";
+      nodes.router = router;
+      nodes.client = { pkgs, ... }: with pkgs.lib; {
+        environment.systemPackages = [ pkgs.iptables ]; # to debug firewall rules
+        virtualisation.vlans = [ 1 ];
+        networking = {
+          useNetworkd = networkd;
+          firewall.logReversePathDrops = true; # to debug firewall rules
+          # reverse path filtering rules for the ipvlan interface seem
+          # to be incorrect, causing the test to fail. Disable temporarily.
+          firewall.checkReversePath = false;
+          firewall.allowPing = true;
+          useDHCP = true;
+          ipvlans.ipvlan = {
+            interface = "eth1";
+            mode = "l2";
+            flags = "bridge";
+          };
+          interfaces.eth1.ipv4.addresses = mkOverride 0 [ ];
+          dhcpcd.extraConfig = ''
+            interface ipvlan
+            broadcast true
+            clientid 00:11:22:33:44:55:66:77:88:99
+          '';
+        };
+        # TODO: Try to remove this when not using networkd? But naively using optionalAttrs and mkIf causes infinite recursion.
+        systemd.network.networks."40-ipvlan" = {
+          matchConfig.Name = "ipvlan";
+          networkConfig = {
+            DHCP = "yes";
+          };
+          dhcpConfig = {
+            RequestBroadcast = true;
+            # NOTE: Why is this required for scripted/dhcpcd but not networkd?
+            # ClientIdentifier = "00:11:22:33:44:55:66:77:88:99";
```

Comment:
I think dhcpcd does use mac addresses while networkd uses duid by default for dhcp.

Reply:
Removed *NOTE* and modified dhcpcd's behaviour to be closer to networkd's.